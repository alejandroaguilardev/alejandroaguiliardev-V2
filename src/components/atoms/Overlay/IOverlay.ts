import { ReactElement } from "react";

export interface IOverlay {
	children: ReactElement | ReactElement[];
}
