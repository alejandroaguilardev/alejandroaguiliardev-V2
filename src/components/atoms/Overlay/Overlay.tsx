import { Box } from "@mui/material"
import { IOverlay } from "./IOverlay"

export const Overlay = ({ children }: IOverlay) => {
    return (
        <Box
            sx={{
                alignItems: "center",
                bottom: 0,
                display: "flex",
                justifyContent: "center",
                left: 0,
                gap:4,
                position: "absolute",
                opacity:0,
                top: 0,
                right: 0,
                transition: '1s all',
                "&:hover": {
                    backgroundColor: "rgba(0,0,0,.7)",
                    opacity:1,
                }
            }}>
            {children}
        </Box>
    )
}
