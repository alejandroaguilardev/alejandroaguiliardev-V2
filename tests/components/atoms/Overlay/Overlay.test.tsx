import { render, screen } from '@testing-library/react';
import { Overlay } from '../../../src/components/atoms/Overlay/Overlay';
describe("Overlay Test", ()=> {
    test("toMacthSnapshot", () => {
        const {container} = render(<Overlay>
            <h1>Overlay</h1>
        </Overlay>)
        expect(container).toMatchSnapshot();
        expect(screen.getByText("Overlay")).toBeTruthy();
    })
})