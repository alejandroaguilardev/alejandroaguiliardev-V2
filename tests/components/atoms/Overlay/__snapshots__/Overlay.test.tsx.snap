// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Test toMacthSnapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-lzhgz5"
  >
    <h1>
      Overlay
    </h1>
  </div>
</div>
`;
